fdbserver/BlobManager.actor.cpp

```diff
@@ -610,107 +675,106 @@ ACTOR Future<Void> doRangeAssignment(Reference<BlobManagerData> bmData,
 	return Void();
 }
 
-ACTOR Future<Void> rangeAssigner(Reference<BlobManagerData> bmData) {
-	loop {
-
-		state RangeAssignment assignment = waitNext(bmData->rangesToAssign.getFuture());
-		state int64_t seqNo = bmData->seqNo;
-		bmData->seqNo++;
-
-		// modify the in-memory assignment data structures, and send request off to worker
-		state UID workerId;
-		if (assignment.isAssign) {
-			bool skip = false;
-			// Ensure range isn't currently assigned anywhere, and there is only 1 intersecting range
-			auto currentAssignments = bmData->workerAssignments.intersectingRanges(assignment.keyRange);
-			int count = 0;
-			for (auto i = currentAssignments.begin(); i != currentAssignments.end(); ++i) {
-				if (assignment.assign.get().type == AssignRequestType::Continue) {
-					ASSERT(assignment.worker.present());
-					if (i.range() != assignment.keyRange || i.cvalue() != assignment.worker.get()) {
-						TEST(true); // BM assignment out of date
-						if (BM_DEBUG) {
-							fmt::print("Out of date re-assign for ({0}, {1}). Assignment must have changed while "
-							           "checking split.\n  Reassign: [{2} - {3}): {4}\n  Existing: [{5} - {6}): {7}\n",
-							           bmData->epoch,
-							           seqNo,
-							           assignment.keyRange.begin.printable(),
-							           assignment.keyRange.end.printable(),
-							           assignment.worker.get().toString().substr(0, 5),
-							           i.begin().printable(),
-							           i.end().printable(),
-							           i.cvalue().toString().substr(0, 5));
-						}
-						skip = true;
+static bool handleRangeAssign(Reference<BlobManagerData> bmData, RangeAssignment assignment) {
+	int64_t seqNo = bmData->seqNo;
+	bmData->seqNo++;
+
+	// modify the in-memory assignment data structures, and send request off to worker
+	UID workerId;
+	if (assignment.isAssign) {
```

Comment:
Nit: a minor style comment, could we split these into 2 more function calls? It seems if (assignment.isAssign) {} is completely isolated. Would save an indentation and be clearer about the 2 paths. Can be done in the future for sure.